plugins/utils/flash.nix

```diff
@@ -6,589 +6,664 @@
-        '';
 
-        pattern = helpers.defaultNullOpts.mkStr "" "initial pattern to use when opening flash";
+          action = helpers.defaultNullOpts.mkLuaFn defaults.action ''
+            action to perform when picking a label.
+            defaults to the jumping logic depending on the mode.
 
-        continue = helpers.defaultNullOpts.mkBool false ''
-          When `true`, flash will try to continue the last search
-        '';
+            @type fun(match:Flash.Match, state:Flash.State)
+          '';
 
-        config = helpers.defaultNullOpts.mkLuaFn "nil" ''
-          Set config to a function to dynamically change the config
-          @type fun(opts:Flash.Config)
-        '';
+          pattern = helpers.defaultNullOpts.mkStr defaults.pattern ''
+            Initial pattern to use when opening flash.
+          '';
 
-        prompt = {
-          enabled = helpers.defaultNullOpts.mkBool true ''
-            options for the floating window that shows the prompt, for regular jumps
+          continue = helpers.defaultNullOpts.mkBool defaults.continue ''
+            When `true`, flash will try to continue the last search.
           '';
 
-          # Not sure what is the type...
-          prefix = helpers.defaultNullOpts.mkListOf types.anything [
-            [
-              "⚡"
-              "FlashPromptIcon"
-            ]
-          ] "";
-          winConfig = helpers.defaultNullOpts.mkAttrsOf types.anything {
-            relative = "editor";
-            width = 1;
-            height = 1;
-            row = -1;
-            col = 0;
-            zindex = 1000;
-          } "See nvim_open_win for more details";
-        };
+          config = helpers.defaultNullOpts.mkLuaFn defaults.config ''
+            Set config to a function to dynamically change the config.
 
-        remoteOp = {
-          restore = helpers.defaultNullOpts.mkBool false ''
-            restore window views and cursor position after doing a remote operation
+            @type fun(opts:Flash.Config)
           '';
 
-          motion = helpers.defaultNullOpts.mkBool false ''
-            For `jump.pos = "range"`, this setting is ignored.
-            - `true`: always enter a new motion when doing a remote operation
-            - `false`: use the window's cursor position and jump target
-            - `nil`: act as `true` for remote windows, `false` for the current window
-          '';
-        };
-      };
-    in
-    helpers.neovim-plugin.extraOptionsOptions
-    // {
-      enable = mkEnableOption "flash.nvim";
+          prompt = {
+            enabled = helpers.defaultNullOpts.mkBool defaults.prompt.enabled ''
+              Options for the floating window that shows the prompt, for regular jumps.
+            '';
 
-      package = helpers.mkPluginPackageOption "flash.nvim" pkgs.vimPlugins.flash-nvim;
+            # Not sure what the type is...
+            # Think it's listOf (maybeRaw (listOf (maybeRaw str)))?
```

Comment:
This needs to be solved.

Reply:
The comment is not mine, it is already present in `main`.

Same as https://github.com/nix-community/nixvim/pull/1714#discussion_r1647191887, I think I'd have to play with actually using/configuring the plugin to understand to correct solution.

The current type will at least allow users who know how to configure the plugin to do so, though they won't get build-time type checking.

Reply:
Ok, I didn't pay attention.